macrame-play/src/test/scala/macrame/play/enums/EnumTest.scala

```diff
@@ -19,9 +19,9 @@ class EnumTest extends FunSuite {
       }
       object Color extends AsJson[Color]
 
-      assert(Json.toJson(Color.Red) == JsString("Red"))
-      assert(Json.toJson(Color.Blue) == JsString("BLUE"))
-      assert(Json.toJson(Color.Yellow) == JsString("YELLOW"))
+      assert(Json.toJson(Color.Red : Color) == JsString("Red"))
+      assert(Json.toJson(Color.Blue : Color) == JsString("BLUE"))
+      assert(Json.toJson(Color.Yellow : Color) == JsString("YELLOW"))
```

Comment:
`Writes[A]` is now invariant (used to be contravariant).

Reply:
That's not great. I understand we don't want `enum` to generate `Writes` instances, but maybe we can get away with it, for example, with

    trait EnumMember {
        protected def memberName: String
    }

    // this part is generated by enum macro
    case object Red extends Color with EnumMember {
        protected def memberName: String = "Red"
    }

    // and this is in macrame-play
    implicit def enumWrites[E :< EnumMember]: Writes[E] = Writes[E] {e =>
        JsString(e.memberName)
    }

I haven't tried that, but you get the idea.

Reply:
Hmm, this actually works. Thanks.